dashboard/app/controllers/test_controller.rb

```diff
@@ -148,6 +148,23 @@ def assign_course_as_student
     head :ok
   end
 
+  def assign_section_to_course_and_unit
+    return unless (teacher = current_user)
+
+    unit_group = UnitGroup.find_by_name(params.require(:course_name))
+    unit_group_unit = unit_group.default_unit_group_units.find_by!(position: params.require(:unit_position))
+    raise "Unit #{params[:unit_position].inspect} not found in course #{params[:course_name].inspect}" unless unit_group_unit
+    unit = unit_group_unit.script
```

Comment:
Consider using the Query function we already have for getting Unit* information given a course name and unit position:
```suggestion
    course_name = params.require(:course_name)
    unit_position = params.require(:unit_position)
    unit_context = Queries::Courses.get_unit_context(course_name, unit_position)
    raise "Unit #{unit_position.inspect} not found in course #{course_name.inspect}" unless unit_context
    unit_group = unit_context[:unit_group]
    unit = unit_context[:unit]
```